submissions/antonkottans/memory-pair-game/script.js

```diff
@@ -0,0 +1,206 @@
+document.addEventListener("DOMContentLoaded", () => {
+    let timerId = undefined;
+
+    const startTimer = () => {
+        document.querySelectorAll(".opened");
+        let minutes = 0;
+        let seconds = 0;
+        let timerOutput = document.querySelector(".timer");
+        const increaseTimer = () => {
+            seconds++;
+            if (seconds === 60) {
+                minutes++;
+                seconds = 0;
+            }
+        };
+        const getFormattedTimer = () => {
+            return `${minutes < 10 ? "0" : ""}${minutes}:${
+                seconds < 10 ? "0" : ""
+            }${seconds}`;
+        };
+        timerId = setInterval(() => {
+            increaseTimer();
+            if (minutes === 60) timerOutput.textContent = "you lost";
+            timerOutput.textContent = getFormattedTimer();
+        }, 1000);
+    };
+
+    const createClickHandler = () => {
+        let openned = 0;
```

Comment:
By storing this value into a closure and not some global scope or a state, you introduced a bug, where someone can open 5 pairs of cards, press "Start" to restart the game, open 1 pair and win, since you're not checking for all open pairs, you're checking for a certain amount of pairs. Either clean this counter on game restart or make a better win condition check. Either way, avoid using closures in such way, it complicates the code and invites mistakes. 

Also `opened` is a name for variable of a boolean type. Try asking your variable name as a question and you'll see if it fits the value type.

_opened? - true_ seems appropriate.
_opened? - 2_ seems confusing.
_openedAmount? - 2_ seems a lot better.